app/controllers/cm_admin/resource_controller.rb

```diff
@@ -0,0 +1,199 @@
+      @ar_object = @model.ar_model.name.classify.constantize.find(params[:id])
+      respond_to do |format|
+        if @ar_object.destroy
+          format.html { redirect_back fallback_location: cm_admin.send("#{@model.name.underscore}_index_path"), notice: "#{action_name.titleize} #{@ar_object.class.name.downcase} is successful" }
+        else
+          format.html { redirect_back fallback_location: cm_admin.send("#{@model.name.underscore}_index_path"), notice: "#{action_name.titleize} #{@ar_object.class.name.downcase} is unsuccessful" }
+        end
+      end
+    end
+
+    def cm_custom_method(params)
+      scoped_model = "CmAdmin::#{@model.name}Policy::Scope".constantize.new(Current.user, @model.name.constantize).resolve
+      resource_identifier
+      respond_to do |format|
+        if @action.action_type == :custom
+          if @action.child_records
+            format.html { render @action.layout }
+          elsif @action.display_type == :page
+            data = @action.parent == "index" ? @ar_object.data : @ar_object
+            format.html { render @action.partial }
+          else
+            ar_object = @action.code_block.call(@ar_object)
+            if ar_object.errors.empty?
+              redirect_url = @model.current_action.redirection_url || @action.redirection_url || request.referrer || "/cm_admin/#{@model.ar_model.table_name}/#{@ar_object.id}"
+              format.html { redirect_to redirect_url, notice: "#{@action.name.titleize} is successful" }
+            else
+              error_messages = ar_object.errors.full_messages.map{|error_message| "<li>#{error_message}</li>"}.join
+              format.html { redirect_to request.referrer, alert: "<b>#{@action.name.titleize} is unsuccessful</b><br /><ul>#{error_messages}</ul>" }
+            end
+          end
+        end
+      end
+    end
+
+    def resource_identifier
+      @ar_object, @associated_model, @associated_ar_object = custom_controller_action(action_name, params.permit!) if !@ar_object.present? && params[:id].present?
+      authorize controller_name.classify.constantize, policy_class: "CmAdmin::#{controller_name.classify}Policy".constantize if defined? "CmAdmin::#{controller_name.classify}Policy".constantize
+      aar_model = request.url.split('/')[-2].classify.constantize  if params[:aar_id]
+      @associated_ar_object = aar_model.find(params[:aar_id]) if params[:aar_id]
+      nested_tables = @model.available_fields[:new].except(:fields).keys
+      nested_tables += @model.available_fields[:edit].except(:fields).keys
+      @reflections = @model.ar_model.reflect_on_all_associations
+      nested_tables.each do |table_name|
+        reflection = @reflections.select {|x| x if x.name == table_name}.first
+        if reflection.macro == :has_many
+          @ar_object.send(table_name).build if action_name == "new" || action_name == "edit"
+        else
+          @ar_object.send(('build_' + table_name.to_s).to_sym) if action_name == "new"
```

Comment:
@anbublacky - This can be a `elsif` condition instead
```suggestion
        elsif action_name == "new"
          @ar_object.send(('build_' + table_name.to_s).to_sym)
```